.eleventy.js

```diff
@@ -0,0 +1,59 @@
+const makeItemLink = (slug) => `#${slug}`
+
+module.exports = function (config) {
+  // Add a filter using the Config API
+  config.addFilter('linkTarget', makeItemLink);
+
+  config.addFilter('linkIfExistsInCollection', (word, collection) => {
+    const existingDefintion = collection.find(item => item.data.title === word)
+
+    if (existingDefintion) {
+      return `<a href=${makeItemLink(existingDefintion.data.slug)}>${word}</a>`
+    }
+
+    return word
+  })
+
+  // just a debug filter to lazily inspect the content of anything in a template
+  config.addFilter('postInspect', function (post) {
+    console.log(post);
+  })
+
+  config.addPassthroughCopy({'_site/css/': 'assets/css/'})
+
+  // Add collections here
+  config.addCollection('definitions', collection => {
+    return [
+      ...collection
+        .getFilteredByGlob('./11ty/definitions/*.md')
+        .sort((a, b) => {
+          // `toLowerCase()` is just a safety measure, slugs should be lower case anyway
+          // `localeCompare()` is super cool: http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare
+          return a.data.slug.toLowerCase().localeCompare(b.data.slug.toLowerCase())
+        })]
+  })
+
+  config.addCollection('definedWords', collection => {
+    return [
+      ...collection
```

Comment:
Oh interesting use of spread here, did you encounter a bug from this? I thought we created new arrays when you call `getFilteredByGlob` so that you could mutate them with sort. Specifically this use of `filter` creates a new array: https://github.com/11ty/eleventy/blob/a0ee45331909b71b30c28332a58fdcc33fe8566a/src/TemplateCollection.js#L43

Reply:
To be honest, I saw that in the source code of @hankchizljaw’s Hylia project and ran with it. Works equally well without the spread. So, I am going to remove that.